themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx

```diff
@@ -68,116 +92,202 @@ class SigningInPage extends React.Component<SigningInPageProps, SigningInPageSta
     public constructor(props: SigningInPageProps) {
         super(props);
         this.state = {
-            twoFactorEnabled: true,
-            twoFactorEnabledText: Msg.localize('twoFactorEnabled'),
-            isTotpConfigured: features.isTotpConfigured,
+            credentialContainers: new Map()
         }
-        this.setLastPwdUpdate();
+
+        this.getCredentialContainers();
     }
 
-    private setLastPwdUpdate(): void {
-        AccountServiceClient.Instance.doGet("/credentials/password")
-            .then((response: AxiosResponse<PasswordDetails>) => {
-                if (response.data.lastUpdate) {
-                    const lastUpdate: number = response.data.lastUpdate;
-                    this.setState({lastPasswordUpdate: lastUpdate});
-                }
+    private getCredentialContainers(): void {
+        AccountServiceClient.Instance.doGet("/credentials")
+            .then((response: AxiosResponse<CredentialContainer[]>) => {
+
+                const allContainers: CredContainerMap = new Map();
+                response.data.forEach(container => {
+                    let categoryMap = allContainers.get(container.category);
+                    if (!categoryMap) {
+                        categoryMap = new Map();
+                        allContainers.set(container.category, categoryMap);
+                    }
+                    categoryMap.set(container.type, container);
+                });
+
+                this.setState({credentialContainers: allContainers});
             });
     }
     
-    private handleTwoFactorSwitch = () => {
-        if (this.state.twoFactorEnabled) {
-            this.setState({twoFactorEnabled: false, twoFactorEnabledText: Msg.localize('twoFactorDisabled')})
-        } else {
-            this.setState({twoFactorEnabled: true, twoFactorEnabledText:  Msg.localize('twoFactorEnabled')})
-        }
-    }
-
-    private handleRemoveTOTP = () => {
-      AccountServiceClient.Instance.doDelete("/totp/remove")
+    private handleRemove = (credentialId: string, userLabel: string) => {
+      AccountServiceClient.Instance.doDelete("/credentials/" + credentialId)
         .then(() => {
-            this.setState({isTotpConfigured: false});
-            ContentAlert.success('successTotpRemovedMessage');
+            this.getCredentialContainers();
+            ContentAlert.success('successRemovedMessage', [userLabel]);
         });
     }
 
     public render(): React.ReactNode {
```

Comment:
When a credential is not set up, the info text is not vertically aligned.
![Screenshot 2020-01-22 at 17 31 05](https://user-images.githubusercontent.com/13906855/72916685-bf02f780-3d42-11ea-8999-d10866f49cb5.png)

Reply:
I'm not sure that it needs to be aligned.  It is centered instead, which is standard for Empty State.

I considered using the EmptyState component for this, which makes it a little fancier.  With that, it becomes much more obvious that it needs to be centered.

Reply:
What I meant was that e.g. "Created: {date}" is vertically aligned and it looks great whereas "Credential is not set up." is not. But thinking about this more, I think it can be either way (centered or not centered).